### PR TITLE
GCW-2414 read unread tracking of messages

### PR DESCRIPTION
**Pull request by abhinav098**
Hi team, this PR is for marking unread messages to read. 
Please review
Thanks

### CODE REVIEW

File: app/controllers/subscription.js [hunk 1]
```diff
@@ -201,6 +202,39 @@ export default Ember.Controller.extend({
         this.updateCartAvailability(0, cartItem);
       }
     }
+
+    // mark message read here
+    if (type === "message") {
```
Comment (patrixr) [inline app/controllers/subscription.js]: what's the alternative to `message` ? 🤔

File: app/controllers/subscription.js [hunk 2]
```diff
@@ -201,6 +202,39 @@ export default Ember.Controller.extend({
...
+      var router = this.get("target");
+      var currentUrl = window.location.href.split("#").get("lastObject");
+
+      var messageRoute = this.get("messagesUtil").getRoute(data.item[type]);
```
Comment (patrixr) [inline app/controllers/subscription.js]: no real problem, but this method is really huge and complicated 😱 , if there's room for some refactoring into single-purpose methods that'd be a good improvement.

File: app/services/messages.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import Ember from "ember";
```
Comment (patrixr) [inline app/services/messages.js]: Good call with a service, we need as much logic to be centralized 👍

Comment (patrixr) [reply app/services/messages.js]: Actually, if there's other pieces of code that can be added to the service, that'd be a good way to go

File: app/controllers/orders/detail.js [hunk 4]
```diff
@@ -6,22 +6,13 @@ import AjaxPromise from 'browse/utils/ajax-promise';
 export default applicationController.extend({
   order: Ember.computed.alias('model.order'),
   showCancelBookingPopUp: false,
-  orderSummaryTabs: [{
```
Comment (patrixr) [inline app/controllers/orders/detail.js]: isn't that needed for the tabs ? 😱

Comment (abhinav098) [reply app/controllers/orders/detail.js]: nope, not needed now, code has been changed.

File: app/services/messages.js (hunk 3 above)
Comment (abhinav098) [reply app/services/messages.js]: sure

File: app/controllers/subscription.js (hunk 1 above)
Comment (abhinav098) [reply app/controllers/subscription.js]: there are many, order, item, version etc.

File: app/controllers/orders/detail.js (hunk 4 above)
Comment (patrixr) [reply app/controllers/orders/detail.js]: cool :)

**Pull request merged**